job-service-db/src/main/resources/procedures/getJobs.sql

```diff
@@ -133,7 +140,14 @@ BEGIN
     IF in_offset > 0 THEN
         sql := sql || ' OFFSET ' || in_offset;
     END IF;
-
+    -- Join onto the labels after paging to avoid them bloating the row count
+    sql := sql || ' ) as job LEFT JOIN public.label lbl ON lbl.partition_id = job.partition_id '
+               || 'AND lbl.job_id = job.job_id';
+    IF in_labels IS NOT NULL AND ARRAY_LENGTH(in_labels, 1) > 0 THEN
+        sql := sql || ' INNER JOIN public.label lbl_filter ON lbl_filter.partition_id = job.partition_id '
+                   || ' AND lbl_filter.job_id = job.job_id '
+                   || ' AND lbl_filter.label = ANY(' || quote_literal(in_labels) || ') ';
+    END IF;
```

Comment:
This doesn't look right to me.  You're applying the label restriction after the limit/offset.